### ISSUE TITLE
osx support?

### ISSUE THREAD

**Issue by seanjensengrey** [nathan7/libfringe#35]
I tried building libfringe on OSX  10.10.5 using

```
rustc 1.13.0-nightly (eac41469d 2016-08-30)
cargo 0.13.0-nightly (3e41b6b 2016-08-29)
```

```
Darwin basho.local 14.5.0 Darwin Kernel Version 14.5.0: Thu Jun 16 19:58:21 PDT 2016; root:xnu-2782.50.4~1/RELEASE_X86_64 x86_64
```

# libc::MAP_STACK

```
(x.env) basho:libfringe basho$ cargo test
   Compiling fringe v1.0.2 (file:///Users/basho/w/days/rust-fringe/libfringe)
error[E0425]: unresolved name `libc::MAP_STACK`
  --> src/os/sys.rs:21:28
   |
21 | const STACK_FLAGS: c_int = libc::MAP_STACK
   |                            ^^^^^^^^^^^^^^^

```

then I modified `src/os/sys.rs` to include `macos` in the same set as FreeBSD and Dragonfly. Then  I got

```
(x.env) basho:libfringe basho$ cargo test
   Compiling fringe v1.0.2 (file:///Users/basho/w/days/rust-fringe/libfringe)
error: <inline asm>:8:7: error: unknown directive
      .local __morestack
      ^

  --> src/arch/x86_64.rs:54:5
   |
54 |     asm!(
   |     ^

```

**Comment by whitequark** [comment]
Ah, `.local` is an ELF-specific directive... let me see what's there for Mach-O.

**Comment by whitequark** [comment]
Okay, so unfortunately this is not ergonomic *just yet*, because of a bug in ld64 and two bugs in rustc that I've just found.

**Comment by whitequark** [comment]
@seanjensengrey Blocked by https://github.com/rust-lang/rust/issues/36176.

**Comment by seanjensengrey** [comment]
Excellent progress! Watching all three. I'll switch to Linux for my experiments in the mean time.

**Comment by whitequark** [comment]
Actually, nevermind, I fixed this all in #35.
